vizro-core/src/vizro/models/_components/table.py

```diff
@@ -0,0 +1,82 @@
+
+logger = logging.getLogger(__name__)
+
+
+class Table(VizroBaseModel):
+    """Wrapper for table components to visualize in dashboard.
+
+    Args:
+        type (Literal["table"]): Defaults to `"table"`.
+        table (CapturedCallable): Table like object to be displayed. Current choices include:
+            [`dash_table.DataTable`](https://dash.plotly.com/datatable).
+        actions (List[Action]): See [`Action`][vizro.models.Action]. Defaults to `[]`.
+    """
+
+    type: Literal["table"] = "table"
+    table: CapturedCallable = Field(..., description="Table to be visualized on dashboard") #ADD import path
+    actions: List[Action] = []
+
+    # Component properties for actions and interactions
+    _input_property: str = PrivateAttr("active_cell")
+    _output_property: str = PrivateAttr("children")
+
+    # validator
+    set_actions = _action_validator_factory("active_cell")  # type: ignore[pydantic-field]
+
+    @validator("table")
+    def process_component_data_frame(cls, table, values):
+        data_frame = table["data_frame"]
+
+        # Enable running "iris" from the Python API and specification of "data_frame": "iris" through JSON.
+        # In these cases, data already exists in the data manager and just needs to be linked to the component.
+        if isinstance(data_frame, str):
+            data_manager._add_component(values["id"], data_frame)
+            return table
+
+        # Standard case for df: pd.DataFrame.
+        # Extract dataframe from the captured function and put it into the data manager.
+        dataset_name = str(id(data_frame))
+
+        logger.debug("Adding data to data manager for Table with id %s", values["id"])
+        # If the dataset already exists in the data manager then it's not a problem, it just means that we don't need
+        # to duplicate it. Just log the exception for debugging purposes.
+        try:
+            data_manager[dataset_name] = data_frame
+        except ValueError as exc:
+            logger.debug(exc)
+
+        data_manager._add_component(values["id"], dataset_name)
+
+        # No need to keep the data in the captured function any more so remove it to save memory.
+        del table["data_frame"]
+        return table
+
+    # Convenience wrapper/syntactic sugar.
+    def __call__(self, **kwargs):
+        kwargs.setdefault("data_frame", data_manager._get_component_data(self.id))  # type: ignore[arg-type]
+        return self.table(**kwargs)
+
+    # Convenience wrapper/syntactic sugar.
+    def __getitem__(self, arg_name: str):
+        # pydantic discriminated union validation seems to try Table["type"], which throws an error unless we
+        # explicitly redirect it to the correct attribute.
+        if arg_name == "type":
+            return self.type
+        return self.table[arg_name]
+
+    @_log_call
+    def build(self):
+        return html.Div(dash_table.DataTable(pd.DataFrame().to_dict('records'), []), id=self.id)
```

Comment:
This looks weird to me. How can we support the AgGrid now? Isn't that we need to call `self.table._function(...)` in the build method?

Reply:
Ah I see now. This now follows the `figure=create_empty_fig("")` approach in the `vm.Graph.build` method and it really makes sense 👍

Reply:
Yes although there is still a weird behaviour I couldn't immediately figure out. When there are no controls (unlike the `Graph`) it doesn't load the table right. I suppose it doesn't trigger the on page load action?

Reply:
`_on_page_load` targets are all components returned from the `_get_components_with_data` function (so tables should also be included). 

But, the _on_page_load action will be created in the `Page.pre_build` only:
`if any(isinstance(component, Graph) for component in self.components):`

So we probably need to adjust this if statement so that it passes even if there are only table components on the page.

Reply:
This sounds a case where we might want to replace `isinstance(component, Graph)` with `callable(component)` if we like the idea https://github.com/mckinsey/vizro/pull/114#issuecomment-1772454425?

Reply:
Implemented a slight alternative that was quick to do, which follows the idea that we now have `_callable_component`. Can refactor further in the future

Reply:
Changed this to `hasattr(...,"figure")` given that we want to unify Graph and Table to carry this attritbute now